src/regex_split.cpp

```diff
@@ -106,10 +127,33 @@ void RegexSplit::validate_and_infer_types() {
 }
 
 bool RegexSplit::evaluate(ov::TensorVector& outputs, const ov::TensorVector& inputs) const {
-    if (m_pretokenizer == nullptr) {
-        auto split_pattern = std::string(inputs[5].data<const char>(), inputs[5].get_size());
-        m_pretokenizer = std::make_shared<pretokenizers::SplitPreTokenizer>(split_pattern, split_modes.at(m_behaviour), m_invert);
-    };
+    auto split_pattern = std::string(inputs[5].data<const char>(), inputs[5].get_size());
+    compile_pattern_if_necessary(split_pattern);
+    
+    // If RE2 didn't compiled successfully fallback to PCRE2 matcher.
+    std::function<std::optional<std::pair<size_t, size_t>>(const std::string&, size_t)> get_next_match;
+    if (m_search_pattern_re2) {
+        get_next_match = [this](const std::string& str, size_t curr_start) -> std::optional<std::pair<size_t, size_t>>{
```

Comment:
consider moving this int a separate function or method.

Reply:
It's not possible to move into a separate function because they need access to m_search_pattern_re2, m_search_pattern_pcre2 but since they are private they are inaccessable.
And it's preferable avoid adding unnecessary methods to operation.

Therefore let's leave this lambdas as is.